apps/src/StudioApp.js

```diff
@@ -843,17 +868,96 @@ StudioApp.prototype.createModalDialog = function(options) {
   return this.feedback_.createModalDialog(options);
 };
 
+StudioApp.prototype.recordUserViewedHint_ = function (hint, config) {
+  hint.alreadySeen = true;
+
+  authoredHintUtils.recordUnfinishedHint({
+    // level info
+    script_id: config.scriptId,
+    level_id: config.level.scriptLevelId,
+
+    // hint info
+    hint_id: hint.hintId,
+    hint_class: hint.hintClass,
+    hint_type: hint.hintType,
+  });
+};
+
+StudioApp.prototype.onReportComplete = function (response) {
+  authoredHintUtils.finishHints({
+    time: ((new Date().getTime()) - this.initTime),
+    attempt: this.attempts,
+    test_result: this.lastTestResult,
+    activity_id: response && response.activity_id,
+    level_source_id: response && response.level_source_id,
+  });
+};
+
+StudioApp.prototype.showHint_ = function (hint, config) {
+
+  $('#prompt-icon').qtip({
+    content: {
+      text: function(html, api) {
+        var container = document.createElement('div');
+        container.innerHTML = require('./templates/hintSelect.html.ejs')();
+
+        var instructionsButton = container.querySelector('#inst');
+        dom.addClickTouchEvent(instructionsButton, function () {
+          this.showInstructions_(config.level, false);
+          api.destroy();
+        }.bind(this));
+
+        var hintButton = container.querySelector('#newhint');
+        dom.addClickTouchEvent(hintButton, function () {
+          api.set('content.text', hint.content);
+          this.recordUserViewedHint_(hint, config);
+        }.bind(this));
+
+        return container;
```

Comment:
Not 100% sure how well it will play with qtip, but if we limit React to what's currently the container element, we might be okay, i.e.
```
var container = document.createElement('div');
React.render(<HintSelect ...props/>, container);
return container;